woocommerce/woocommerce-ios
# Add activity indicator to TwoColumnHeadlineFootnoteTableViewCell

Improves #11857, from review https://github.com/woocommerce/woocommerce-ios/pull/11914#pullrequestreview-1867228461

## Description
At the moment, when tapping on "See receipt", the loading indicator only appears when we've already pushed the view controller into the navigation stack, but nothing happens in the time between a merchant presses the button and the action is dispatched, which may lead to confusion or multiple tapping.

This PR adds a simple loading state to the existing cell: Upon tapping, the cell starts animating by disabling user interaction, reducing its alpha, and making a loading indicator appear. Once we get a result from the store, we stop animating.

The solution is not perfect, sometimes we might still see a small gap between the loading and the view, but I opted from the simplest approach since improves the current implementation until the next project iteration.

## Testing instructions
1. On a backend-receipt eligible store ( you can use mywootestingstore.mystagingwebsite.com ) open an order that has an existing backend-generated receipt, or create a new order > collect payment by cash (old receipt generation do not have this, since are loaded from device).
2. Tap on the "See receipt" cell
3. Observe that the animation starts, then the receipt web view is pushed into the stack.

## Screenshots

| Normal state | Loading |
|--------|--------|
| <img width="298" alt="Screenshot 2024-02-13 at 10 15 35" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/7a63b3cb-b01f-40f7-b3d4-27560babb0d3"> | <img width="296" alt="Screenshot 2024-02-13 at 10 15 02" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/137f3f35-4651-40ba-b42b-42baf44f565d"> |
